src/utils/createContextWrapper.js

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+
+/**
+ * Creates new trigger class that injects context into overlay.
+ */
+export default function createContextWrapper(Trigger, propName) {
+  return function (contextTypes) {
+    class ContextWrapper extends React.Component {
+      getChildContext() {
+        return this.props.context;
+      }
+
+      render() {
+        // Strip injected props from below.
+        const {wrapped, ...props} = this.props;
+        delete props.context;
+
+        return React.cloneElement(wrapped, props);
+      }
+    }
+    ContextWrapper.childContextTypes = contextTypes;
+
+    class TriggerWithContext {
+      render() {
+        const props = {...this.props};
+        props[propName] = this.getWrappedOverlay();
+
+        return (
+          <Trigger {...props}>
+            {this.props.children}
+          </Trigger>
+        );
+      }
+
+      getWrappedOverlay() {
+        return (
+          <ContextWrapper
+            context={this.context}
+            wrapped={this.props[propName]}
```

Comment:
Again. I got it wrong. Sorry.
I'd seen it as `this.props['propName']`.
It seems I was sleeping yet at morning. :blush: 

Thank you for your patience with my wrong questions :cherries: